src/merkle_tree.rs

```diff
@@ -0,0 +1,221 @@
+use hmac_sha256::Hash;
+
+pub struct MerkleTree {
+    pub root: Option<[u8; 32]>,
+    pub leafs: Vec<[u8; 32]>,
```

Comment:
Just a typo, it should be `leaves` instead of `leafs`.

Reply:
yikes, you're right.